GPU/Workflow/helper/src/GPUWorkflowHelper.cxx

```diff
@@ -109,23 +131,31 @@ std::shared_ptr<const GPUWorkflowHelper::tmpDataContainer> GPUWorkflowHelper::fi
       const auto& tpcTracksMC = recoCont.getTPCTracksMCLabels();
       ioPtr.outputTracksTPCO2MC = tpcTracksMC.data();
     }
+    if (ioPtr.nItsTracks && ioPtr.nTracksTPCITSO2) {
+      retVal->tpcLinkITS.resize(ioPtr.nOutputTracksTPCO2, -1);
+      ioPtr.tpcLinkITS = retVal->tpcLinkITS.data();
+    }
+    if (ioPtr.nTOFClusters && (ioPtr.nTOFMatches || ioPtr.nTPCTOFMatches)) {
+      retVal->tpcLinkTRD.resize(ioPtr.nOutputTracksTPCO2, -1);
+      ioPtr.tpcLinkTRD = retVal->tpcLinkTRD.data();
+    }
+    if (ioPtr.nTRDTracksITSTPCTRD || ioPtr.nTRDTracksTPCTRD) {
+      retVal->tpcLinkTOF.resize(ioPtr.nOutputTracksTPCO2, -1);
+      ioPtr.tpcLinkTOF = retVal->tpcLinkTOF.data();
+    }
   }
 
-  if (maskCl[GID::TRD]) {
-    // o2::trd::getRecoInputContainer(pc, &ioPtr, &recoCont, useMC); // TODO: use this helper here
-  }
-
-  if (maskTrk[GID::ITSTPCTRD] && ioPtr.nTRDTracksITSTPCTRD == 0) {
-    const auto& trdTracks = recoCont.getITSTPCTRDTracks<o2::trd::TrackTRD>();
-    ioPtr.nTRDTracksITSTPCTRD = trdTracks.size();
-    ioPtr.trdTracksITSTPCTRD = trdTracks.data();
-  }
-
-  if (maskTrk[GID::TPCTRD] && ioPtr.nTRDTracksTPCTRD == 0) {
-    const auto& trdTracks = recoCont.getTPCTRDTracks<o2::trd::TrackTRD>();
-    ioPtr.nTRDTracksTPCTRD = trdTracks.size();
-    ioPtr.trdTracksTPCTRD = trdTracks.data();
-  }
+  auto creator = [&recoCont, &retVal](auto& trk, GID gid, float time, float) {
+    if constexpr (std::is_same_v<std::decay_t<decltype(trk)>, o2::tpc::TrackTPC>) {
+      time = trk.getTime0();
+    }
+    retVal->globalTracks.emplace_back(&trk);
```

Comment:
It is a bit unfortunate that I have to use the `std::is_same_v<std::decay_t<decltype(trk)>, o2::tpc::TrackTPC>` here.
I would have prefered something like `if constexpr (ID == GID::TPC) {`, but that doesn't work, since the function argument cannot be constexpr, and I also found now way to pass in template parameters, will check again, perhaps this can be improved.